internal/core/application/service.go

```diff
@@ -3124,13 +3151,21 @@ func (s *service) startWatchingVtxos(vtxos []domain.Vtxo) error {
 		return err
 	}
 
+	if len(scripts) <= 0 {
+		return nil
+	}
+
 	return s.scanner.WatchScripts(context.Background(), scripts)
 }
 
-func (s *service) stopWatchingVtxos(vtxos []domain.Vtxo) {
-	scripts, err := s.extractVtxosScriptsForScanner(vtxos)
-	if err != nil {
-		log.WithError(err).Warn("failed to extract scripts from vtxos")
+func (s *service) stopWatchingVtxos(tapkeys []string) {
+	scripts := make([]string, 0, len(tapkeys))
+	for _, key := range tapkeys {
+		// script = OP_1 OP_PUSHBYTES_32 <key>
+		scripts = append(scripts, "5120"+key)
```

Comment:
let's keep using fmt.Sprintf to concat:
```suggestion
		scripts = append(scripts, fmt.Sprintf("5120%s",key))
```